velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -482,6 +554,161 @@ struct DateFromUnixDateFunction {
   }
 };
 
+template <typename T>
+struct DateTruncFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  const tz::TimeZone* timeZone_ = nullptr;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* formatString,
+      const arg_type<Timestamp>* /*timestamp*/) {
+    timeZone_ = getTimeZoneFromConfig(config);
+  }
+
+  FOLLY_ALWAYS_INLINE void adjustDateTime(
+      std::tm& dateTime,
+      const DateTimeFormat& format) {
+    switch (format) {
```

Comment:
Ditto. Most code are the same and we can add a function in the lib for the reuse of both Presto and Spark.